Src/FluentAssertions/Numeric/NumericAssertions.cs

```diff
@@ -481,5 +481,27 @@ public AndConstraint<TAssertions> NotBeOfType(Type unexpectedType, string becaus
             throw new NotSupportedException("Calling Equals on Assertion classes is not supported.");
 
         private protected virtual bool IsNaN(T value) => false;
+
+        /// <summary>
+        /// A method to generate additional information upon comparison failures.
+        /// </summary>
+        /// <param name="expected">The value to compare the current numeric value with.</param>
+        /// <returns>
+        /// Returns the difference between a number value and the <paramref name="expected" /> value.
+        /// Returns `null` if the compared numbers are small enough that a difference message is irrelevant.
+        /// </returns>
+        private protected virtual T? CalculateDifferenceForFailureMessage(T expected) => null;
+
+        private string GenerateDifferenceMessage(T? expected)
+        {
+            const string noDifferenceMessage = ".";
+            if (!Subject.HasValue || !expected.HasValue)
+            {
+                return noDifferenceMessage;
+            }
+
+            var difference = CalculateDifferenceForFailureMessage(expected.Value);
+            return difference is null ? noDifferenceMessage : $" (difference of {difference}).";
```

Comment:
`{difference}` is formatted using `CurrentCulture`.

If you add `using static System.FormattableString;` to the usings, you can write

```cs
return difference is null ? noDifferenceMessage : Invariant($" (difference of {difference}).");
```